node/bft/src/primary.rs

```diff
@@ -1560,14 +1560,17 @@ impl<N: Network> Primary<N> {
             self.try_increment_to_the_next_round(batch_round).await?;
         }
 
+        // Ensure the primary has all of the transmissions.
+        let transmissions_promise = self.fetch_missing_transmissions(peer_ip, batch_header);        
```

Comment:
Ah that's right. An `.await` is necessary to tell the async runtime to poll the future, which then calls the underlying function.

Reply:
yes, that would work, though since errors trigger early returns, [`try_join`](https://docs.rs/tokio/latest/tokio/macro.try_join.html) should be used instead

Reply:
Thx gents, looks much cleaner too: d0b93272d